enzy_htp/structure/structure.py

```diff
@@ -88,100 +90,189 @@
         4
 
     Saving the structure:
-        >>> structure.to_pdb( "/path/to/copy/of/pdb" )
+        >>> structure.to_pdb( '/path/to/copy/of/pdb' )
 
 Author: Qianzhen (QZ) Shao <qianzhen.shao@vanderbilt.edu>
 Author: Chris Jurich <chris.jurich@vanderbilt.edu>
 Date: 2022-04-03
-"""
+'''
 #TODO(CJ): add a method for changing/accessing a specific residue
 from __future__ import annotations
+import itertools
+import os
 
 import string
 from copy import deepcopy
 import sys
-from typing import List, Set, Dict, Tuple
+from typing import List, Set, Dict, Tuple, Union
 from collections import defaultdict
 
 from enzy_htp.core import _LOGGER
 from enzy_htp.core import file_system as fs
-from enzy_htp.chemical import convert_to_one_letter
+from enzy_htp.chemical import convert_to_one_letter, ResidueType
+from enzy_htp.core.doubly_linked_tree import DoubleLinkNode
 
 from .atom import Atom
 from . import Chain
 from .residue import Residue
-from .metal_atom import MetalAtom
+from .metal_atom import MetalUnit
 from .ligand import Ligand, residue_to_ligand
 from .solvent import Solvent, residue_to_solvent
-from .metal_atom import MetalAtom
+from .metal_atom import MetalUnit
 
 
-class Structure:
-    """Enzyme structure.
+class Structure(DoubleLinkNode): # TODO implement different copy methods for the doubly linked ds; by default are all shollow copy and references
+    '''Enzyme structure.
```

Comment:
Will it always be enzymes? Or can it also be a regular protein also?

Reply:
It can be a protein in general.